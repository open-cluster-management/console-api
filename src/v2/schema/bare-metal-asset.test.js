/** *****************************************************************************
 * Licensed Materials - Property of Red Hat, Inc.
 * Copyright (c) 2020 Red Hat, Inc.
 ****************************************************************************** */
import supertest from 'supertest';
import server, { GRAPHQL_PATH } from '../index';

describe('BareMetalAsset Resolver', () => {
  test('Correctly Resolves BareMetalAsset Query', (done) => {
    supertest(server)
      .post(GRAPHQL_PATH)
      .send({
        query: `
        {
          bareMetalAssets {
            metadata {
              annotations
              creationTimestamp
              labels
              name
              namespace
              resourceVersion
              selfLink
              status
              uid
            }
            bmc {
              address
              credentialsName
            }
            bootMACAddress
            clusterDeployment {
              name
              namespace
            }
            hardwareProfile
            role
          }
        }
      `,
      })
      .end((err, res) => {
        expect(JSON.parse(res.text)).toMatchSnapshot();
        done();
      });
  });
});
