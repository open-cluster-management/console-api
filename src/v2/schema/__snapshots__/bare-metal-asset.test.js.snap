// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BareMetalAsset Resolver Correctly Resolves BareMetalAsset Query 1`] = `
Object {
  "data": Object {
    "bareMetalAsset": null,
  },
  "errors": Array [
    Object {
      "extensions": Object {
        "code": "INTERNAL_SERVER_ERROR",
        "exception": Object {
          "stacktrace": Array [
            "TypeError: Cannot read property 'bmc' of undefined",
            "    at BareMetalAssetModel.getSingleBareMetalAsset (/Users/fxiang/workspaces/console-api/src/v2/models/bare-metal-asset.js:64:33)",
            "    at processTicksAndRejections (internal/process/task_queues.js:97:5)",
          ],
        },
      },
      "locations": Array [
        Object {
          "column": 11,
          "line": 3,
        },
      ],
      "message": "Cannot read property 'bmc' of undefined",
      "path": Array [
        "bareMetalAsset",
      ],
    },
  ],
}
`;

exports[`BareMetalAsset Resolver Correctly Resolves BareMetalAssetSubresources Query 1`] = `
Object {
  "data": Object {
    "bareMetalAssetSubresources": null,
  },
  "errors": Array [
    Object {
      "extensions": Object {
        "code": "INTERNAL_SERVER_ERROR",
        "exception": Object {
          "stacktrace": Array [
            "TypeError: Cannot read property 'bmc' of undefined",
            "    at BareMetalAssetModel.getSingleBareMetalAsset (/Users/fxiang/workspaces/console-api/src/v2/models/bare-metal-asset.js:64:33)",
            "    at processTicksAndRejections (internal/process/task_queues.js:97:5)",
            "    at async Promise.all (index 1)",
            "    at BareMetalAssetModel.getBareMetalAssetSubresources (/Users/fxiang/workspaces/console-api/src/v2/models/bare-metal-asset.js:228:42)",
          ],
        },
      },
      "locations": Array [
        Object {
          "column": 11,
          "line": 3,
        },
      ],
      "message": "Cannot read property 'bmc' of undefined",
      "path": Array [
        "bareMetalAssetSubresources",
      ],
    },
  ],
}
`;

exports[`BareMetalAsset Resolver Correctly Resolves BareMetalAssets Query 1`] = `
Object {
  "data": Object {
    "bareMetalAssets": Array [],
  },
}
`;

exports[`BareMetalAsset Resolver Correctly Resolves CreateBareMetalAsset Query 1`] = `
Object {
  "data": Object {
    "createBareMetalAsset": Object {
      "bmaResult": Object {
        "apiVersion": "v1",
        "code": 404,
        "details": Object {
          "kind": "namespaces",
          "name": "fake-cluster",
        },
        "kind": "Status",
        "message": "namespaces \\"fake-cluster\\" not found",
        "metadata": Object {},
        "reason": "NotFound",
        "status": "Failure",
      },
      "patchedSecretResult": Object {
        "apiVersion": "v1",
        "code": 405,
        "details": Object {},
        "kind": "Status",
        "message": "the server does not allow this method on the requested resource",
        "metadata": Object {},
        "reason": "MethodNotAllowed",
        "status": "Failure",
      },
      "secretResult": Object {
        "apiVersion": "v1",
        "code": 404,
        "details": Object {
          "kind": "namespaces",
          "name": "fake-cluster",
        },
        "kind": "Status",
        "message": "namespaces \\"fake-cluster\\" not found",
        "metadata": Object {},
        "reason": "NotFound",
        "status": "Failure",
      },
      "statusCode": 404,
    },
  },
}
`;

exports[`BareMetalAsset Resolver Correctly Resolves DeleteBareMetalAsset Query 1`] = `
Object {
  "data": Object {
    "deleteBareMetalAssets": Object {
      "errors": Array [
        Object {
          "message": "baremetalassets.inventory.open-cluster-management.io \\"undefined\\" not found",
          "statusCode": 404,
        },
      ],
      "message": "Failed to delete 1 bare metal asset(s)",
      "statusCode": 500,
    },
  },
}
`;

exports[`BareMetalAsset Resolver Correctly Resolves UpdateBareMetalAsset Query 1`] = `
Object {
  "data": Object {
    "updateBareMetalAsset": null,
  },
  "errors": Array [
    Object {
      "extensions": Object {
        "code": "INTERNAL_SERVER_ERROR",
        "exception": Object {
          "stacktrace": Array [
            "TypeError: Cannot read property 'bmc' of undefined",
            "    at BareMetalAssetModel.updateBareMetalAsset (/Users/fxiang/workspaces/console-api/src/v2/models/bare-metal-asset.js:402:31)",
            "    at processTicksAndRejections (internal/process/task_queues.js:97:5)",
          ],
        },
      },
      "locations": Array [
        Object {
          "column": 11,
          "line": 3,
        },
      ],
      "message": "Cannot read property 'bmc' of undefined",
      "path": Array [
        "updateBareMetalAsset",
      ],
    },
  ],
}
`;
